malariagen_data/anoph/cnv_frq.py

```diff
@@ -0,0 +1,698 @@
+
+        debug("compute frequency")
+        with np.errstate(divide="ignore", invalid="ignore"):
+            # ignore division warnings
+            frequency = np.where(nobs > 0, count / nobs, np.nan)
+
+        debug("make dataframe of variants")
+        with warnings.catch_warnings():
+            # ignore "All-NaN slice encountered" warnings
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            max_af = np.nanmax(frequency, axis=1)
+        df_variants = pd.DataFrame(
+            {
+                "contig": region.contig,
+                "start": np.repeat(ds_cnv["gene_start"].values, 2),
+                "end": np.repeat(ds_cnv["gene_end"].values, 2),
+                "windows": np.repeat(ds_cnv["gene_windows"].values, 2),
+                # alternate amplification and deletion
+                "cnv_type": np.tile(np.array(["amp", "del"]), n_genes),
+                "max_af": max_af,
+                "gene_id": np.repeat(ds_cnv["gene_id"].values, 2),
+                "gene_name": np.repeat(ds_cnv["gene_name"].values, 2),
+                "gene_strand": np.repeat(ds_cnv["gene_strand"].values, 2),
+            }
+        )
+
+        debug("add variant label")
+        df_variants["label"] = pandas_apply(
+            self._make_gene_cnv_label,
+            df_variants,
+            columns=["gene_id", "gene_name", "cnv_type"],
+        )
+
+        debug("build the output dataset")
+        ds_out = xr.Dataset()
+
+        debug("cohort variables")
+        for coh_col in df_cohorts.columns:
+            ds_out[f"cohort_{coh_col}"] = "cohorts", df_cohorts[coh_col]
+
+        debug("variant variables")
+        for snp_col in df_variants.columns:
+            ds_out[f"variant_{snp_col}"] = "variants", df_variants[snp_col]
+
+        debug("event variables")
+        ds_out["event_count"] = ("variants", "cohorts"), count
+        ds_out["event_nobs"] = ("variants", "cohorts"), nobs
+        ds_out["event_frequency"] = ("variants", "cohorts"), frequency
+
+        debug("deal with invariants")
+        if drop_invariant:
+            loc_variant = df_variants["max_af"].values > 0
+            ds_out = ds_out.isel(variants=loc_variant)
+            df_variants = df_variants.loc[loc_variant].reset_index(drop=True)
+
+        debug("apply variant query")
+        if variant_query is not None:
+            loc_variants = df_variants.eval(variant_query).values
+            ds_out = ds_out.isel(variants=loc_variants)
+
+        debug("add confidence intervals")
+        _add_frequency_ci(ds=ds_out, ci_method=ci_method)
+
+        debug("tidy up display by sorting variables")
+        ds_out = ds_out[sorted(ds_out)]
+
+        return ds_out
+
+    @staticmethod
+    def _make_gene_cnv_label(gene_id, gene_name, cnv_type):
+        label = gene_id
+        if isinstance(gene_name, str):
+            label += f" ({gene_name})"
+        label += f" {cnv_type}"
+        return label
```

Comment:
```suggestion

def _make_gene_cnv_label(gene_id, gene_name, cnv_type):
    label = gene_id
    if isinstance(gene_name, str):
        label += f" ({gene_name})"
    label += f" {cnv_type}"
    return label
```

No need for this to be a static method, can just be a plain function?